core/src/test/scala/integration/kafka/api/PlaintextAdminIntegrationTest.scala

```diff
@@ -2063,10 +2063,21 @@ class PlaintextAdminIntegrationTest extends BaseAdminIntegrationTest {
     client = Admin.create(createConfig)
 
     val loggerConfig = describeBrokerLoggers()
-    val rootLogLevel = loggerConfig.get(Log4jController.ROOT_LOGGER).value()
-    val logCleanerLogLevelConfig = loggerConfig.get("kafka.cluster.Replica")
-    assertEquals(rootLogLevel, logCleanerLogLevelConfig.value()) // we expect an undefined log level to be the same as the root logger
-    assertEquals("kafka.cluster.Replica", logCleanerLogLevelConfig.name())
+
+    // Root logger is always contained in the loggers list.
+    assertEquals("OFF", loggerConfig.get(Log4jController.ROOT_LOGGER).value())
+
+    // Logger name can't be empty.
+    assertNull(loggerConfig.get(""))
+
+    // Since `kafka` is not defined, it is not contained in the loggers list.
+    assertNull(loggerConfig.get("kafka"))
+
+    // We expect the logger level is inherited from its parent logger level configuration.
+    // For example, `kafka.cluster.Partition` from `kafka` (ERROR).
+    val logCleanerLogLevelConfig = loggerConfig.get("kafka.cluster.Partition")
+    assertEquals("ERROR", logCleanerLogLevelConfig.value()) // we expect an undefined log level to be the same as its parent logger
+    assertEquals("kafka.cluster.Partition", logCleanerLogLevelConfig.name())
```

Comment:
We probably need better coverage in the alter case (`testIncrementalAlterConfigsForLog4jLogLevels()`, below). It tests inheritance from the root logger, but not an ancestor logger. But I guess this is something I should add to my PR.

Reply:
Agree. Let's add a additional test about that case.